bandit/core/manager.py

```diff
@@ -315,8 +320,8 @@ def _parse_file(self, fname, fdata, new_files_list):
             #                                         for the line
             nosec_lines = dict()
             try:
-                fdata.seek(0)
-                tokens = tokenize.tokenize(fdata.readline)
+                buf_data = io.BytesIO(data)
```

Comment:
So, I'm a bit perplexed by what we're doing here, honestly. If `fdata` is always seekable (which, even with your current unmodified changes it is), why are we:

1. Loading an entire file into memory
2. Creating a new buffer for that data
3. Counting the lines the way we are for our metrics?
4. Parsing `nosec` comments in this `try` block when all we really care about is the `TokenError` and `seek` shouldn't be raising that either?

I'm just worried about what happens here with a 10kloc file with 100-200 line length limits

Reply:
This code is all a bit inefficient. It reads the file once to count the number of lines. Then reads the full file again to tokenize in order to retrieve comments like # nosec.

So this bit of code was a bit perplexing to me as well. So stdin is not a seekable file descriptor (fdata.seek(0) will fail).  On line 315, the data is read to its fullest (although I question if that is the case on all platforms).  

But there's probably a better way to do all this.

Reply:
Seems Bandit is iterating through the given file at least 3 times. 
1. to count the lines-of-code in metrics count_locs
2. to tokenize and identify lines marked with # nosec
3. to ast parse the file

Performance would probably be greatly improved on large files if we reduced to a single pass.